### PR TITLE
[SYCL][Bindless] Mangle image handle depending on OS

### PR DESCRIPTION
**Pull request by ProGTX**
On Linux `unsigned long` needs to be `m`, on Windows `y`, when generating PTX builtins in libclc.
Added CMake logic to set the appropriate OS macro.

### CODE REVIEW

File: libclc/ptx-nvidiacl/libspirv/images/image.cl [hunk 1]
```diff
@@ -17,6 +17,14 @@
 #pragma OPENCL EXTENSION cl_khr_3d_image_writes : enable
 #endif
 
+#ifdef _WIN32
+#define MANGLE_FUNC_IMG_HANDLE(namelength, name, prefix, postfix)              \
+  _Z##namelength##name##prefix##y##postfix
+#else
+#define MANGLE_FUNC_IMG_HANDLE(namelength, name, prefix, postfix)              \
+  _Z##namelength##name##prefix##m##postfix
```
Comment (steffenlarsen) [inline libclc/ptx-nvidiacl/libspirv/images/image.cl:25]: I would have expected the remangler to handle this case appropriately. @jchlanda - Can you think of a reason that wouldn't be the case?

**Review by przemektmalon — APPROVED** [review]

File: libclc/ptx-nvidiacl/libspirv/images/image.cl (hunk 1 above)
Comment (jchlanda) [reply libclc/ptx-nvidiacl/libspirv/images/image.cl:25]: Maybe a bug in the remangler, let me have a quick look please.

Comment (jchlanda) [reply libclc/ptx-nvidiacl/libspirv/images/image.cl:25]: We've discussed this with @ProGTX and it turns out that images introduce another case to the remangler that has not been covered previously. The functions can bring in `[u]int64_t` that in turn can be handled (depending on the OS) as `[unsigned] long long` or `[unsigned] long int`, perhaps the best way forward would be to go ahead with this patch?

Comment (steffenlarsen) [reply libclc/ptx-nvidiacl/libspirv/images/image.cl:25]: I am okay with it as a temporary solution, but could you please open an issue about adding the functionality to the remangler so we don't forget about it?

Comment (jchlanda) [reply libclc/ptx-nvidiacl/libspirv/images/image.cl:25]: Sure think: https://github.com/intel/llvm/issues/11082

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by ProGTX** [comment]
@intel/llvm-gatekeepers ready to merge

**Pull request merged**